### PR TITLE
feat(form): textarea component

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Added FormTextArea component
- [x] Added FormTextArea component tests
- [x] Changed InputWrapper to FieldWrapper
- [x] Moved wrapper, prefix, and suffix to shared folder

### CODE REVIEW

File: packages/paste-core/components/form/src/FormTextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        </TextAreaElement>
+        <Suffix>{insertAfter}</Suffix>
+      </FieldWrapper>
+    );
+  }
+);
+
+FormTextArea.displayName = 'FormTextArea';
+
+FormTextArea.propTypes = {
```
Comment (TheSisb) [inline packages/paste-core/components/form/src/FormTextArea.tsx]: please wrap in `if (process.env.NODE_ENV === 'development') {`

File: packages/paste-core/components/form/src/shared/FieldWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      _focusWithin={{
+        boxShadow: 'shadowFocus',
+      }}
+    >
+      {children}
+    </Box>
+  );
+};
+
+FieldWrapper.displayName = 'FieldWrapper';
```
Comment (TheSisb) [inline packages/paste-core/components/form/src/shared/FieldWrapper.tsx]: missing propTypes

File: packages/paste-core/components/form/src/shared/Prefix.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+
+const Prefix: React.FC = ({children}) => {
+  if (children == null) return null;
+  return (
+    <Box
+      display="flex"
+      alignItems="flex-start"
```
Comment (TheSisb) [inline packages/paste-core/components/form/src/shared/Prefix.tsx]: I think if we remove this line we should get the same outcome with less output CSS

File: packages/paste-core/components/form/src/shared/Suffix.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <Box
+      display="flex"
+      alignItems="flex-start"
+      backgroundColor="colorBackground"
+      padding="space30"
+      borderLeftWidth="borderWidth10"
+      borderLeftColor="colorBorderLight"
+      borderLeftStyle="solid"
+      borderTopRightRadius="borderRadius20"
+      borderBottomRightRadius="borderRadius20"
```
Comment (TheSisb) [inline packages/paste-core/components/form/src/shared/Suffix.tsx]: Good fix 👍

File: packages/paste-core/components/form/__tests__/formTextarea.test.tsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+  it('it should render with readOnly', (): void => {
+    const tree = renderer
+      .create(
+        <Theme.Provider theme="console">
+          <FormTextArea id="textarea" onChange={NOOP} readOnly />
+        </Theme.Provider>
+      )
+      .toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (TheSisb) [inline packages/paste-core/components/form/__tests__/formTextarea.test.tsx]: prefer enzyme to validate these props over snapshots

File: packages/paste-core/components/form/src/shared/Prefix.tsx (hunk 3 above)
Comment (richbachman) [reply packages/paste-core/components/form/src/shared/Prefix.tsx]: We do need to set that here, other wise in certain cases the child element will span the full height so the click area is huge. See focus in screenshot:

image

File: packages/paste-core/components/form/__tests__/formTextarea.test.tsx (hunk 5 above)
Comment (SiTaggart) [reply packages/paste-core/components/form/__tests__/formTextarea.test.tsx]: Potentially if you're doing the enzyme test down at https://github.com/twilio-labs/paste/pull/345/files#diff-60403f775a91aa64baef1b2aed75204fR124 you can probably just remove this snapshot test entirely

**Comment by SiTaggart** [comment]
If you're including the auto height stuff then I'll hold off on approval. Otherwise this looks pretty great

File: packages/paste-core/components/form/__tests__/formTextarea.test.tsx (hunk 5 above)
Comment (richbachman) [reply packages/paste-core/components/form/__tests__/formTextarea.test.tsx]: Removed and just left the initial render snapshot.

**Comment by richbachman** [comment]
@SiTaggart decided to move the height sizing to a new ticket so we figure out the proper solution.

File: packages/paste-core/components/form/stories/index.stories.tsx [hunk 6]
```diff
@@ -190,4 +190,33 @@ storiesOf('Components|Form', module)
         <FormHelperText>Info that helps a user with this field.</FormHelperText>
       </>
     );
+  })
+  .add('Textarea Options', () => {
```
Comment (SiTaggart) [inline packages/paste-core/components/form/stories/index.stories.tsx]: Can we get a story for every visual variant and state a textarea can appear in please? Just for VRT purposes.

Comment (richbachman) [reply packages/paste-core/components/form/stories/index.stories.tsx]: @SiTaggart added VRT stories.

**Pull request merged**